charms/istio-pilot/src/charm.py

```diff
@@ -843,8 +1003,7 @@ def _use_https(self):
                 f"Missing {missing}, cannot configure TLS",
                 BlockedStatus,
             )
-        if self._cert_handler.cert and self._cert_handler.key:
-            return True
+        return True if self._cert_handler.cert and self._cert_handler.key else False
```

Comment:
Is there a reason why this:
* returns `False` if both `cert` and `key` are unset
* goes to blocked if one of `cert` and `key` are unset

?

Having `cert_handler.enabled==True` and `cert`/`key` unset sounds like its still a problem.  In this case, TLS is requested but not configured.  Returning `False` here means the gateway will be set up without auth, which probably isn't what is desired

Reply:
Will update, there is no reason to return False here.